### ISSUE TITLE
Replace thaumic energistics with a fork

### ISSUE THREAD

**Issue by realthecaffiend** [Divine-Journey-2/main#567]
2.8.0

**Describe the request:**
Add a fork of thaumic energistics instead; Suggested https://github.com/811Alex/ThaumicEnergistics 

**Reasons why this should be added:**
I am just breaking into thaumcraft more, and realized that the used version of thaumic energistics has some concerns. There are some complaints I've seen in the discord of some instability and buggy behaviour using the terminal (I haven't ran into this yet, but it seems salient to mention), but also: Arcane crafting is just entirely missing. There is at least one fork that has re-implemented this feature linked above (With a large PR back to thaumic energistics with this and a wide variety of fixes and QoL, which seems unlikely to be accepted as the original seems entirely inactive for years). I haven't tested this yet, but this seems like an excellent option; Presumably, the default recipes would need to be altered, quests added, etc, so it's not a trivial addition but would be great quality. I'll update back here if I do any exhaustive testing.

**Describe alternatives you've considered:**
There simply isn't a way to automate arcane crafting (Due to thaumcrafts tie in to research) for the majority of crafts. Some simple ones you could maybe hack up something awful with a table and.. Integrated Dynamics? Right now, I'm using an interface into a chest beside an arcane craft table and it's not at all great. 

**Additional context:**
This would be great, likely later than the original quests, to be able to fully integrate these crafts.

**Comment by WaitingIdly** [comment]
We are aware of that fork. That fork is not accepted in curseforge packs.

**Comment by realthecaffiend** [comment]
Testing notes:

Environment:
Windows 10, jre 1.8.0_161, 4-8GB memory allocation in multimc, 16 GB system
Base environment is Divine Journey 2.8.0, with the following changes, in rough order of how important I suspect they could be:
Added mod: Randomthings MC1.12.2-4.2.7.4 (Content tweaker script to remove every single item, then re-add time in a bottle)
Removed mod: featured servers
Thaumcraft config change: """wuss mode""" enabled (My players and I simply aren't interested in that, after so many times through TC). This doesn't *seem* like it should change any of the results below but just in case
Local client settings have been tinkered with a lot: Visual settings, keybindings, etc. These aren't tracked and shouldn't(?) have an impact.

I cloned this environment in multiMC, and downloaded a recent copy of my server world backup (A few players, but my team of two is the furthest ahead by far, and just starting chapter 20, for a very rough idea of infrastructure). I removed thaumicenergistics 2.2.4, and added thaumicenergistics-v2.2.3-alex811-fork-SNAPSHOT-2. 

Loaded the instance, no more than the standard number of console errors... That I noticed, in the console spam. Load save that was downloaded from server. performed the following quick smoke test:
Standard ME crafting terminal: Items seem to be present, insert/remove works fine. 
JEI search arcane assembler; has an infusion recipe (no auto-infusion, of course)
Open thaumonicon, existing research still present including energistics, mollecular assembler research is now present. Complete research by cheating in parts to craft arcane terminal. 
Note: arcane craft table (not terminal), in same position as on server, same gear, gives a 5% vis discount on some items now: Presence of AE equipment? Did not happen previously on server, including re-crafting items I just recently made
Discover knowledge core in thaumonomicon by infusing up an arcane assembler out of cheated parts
Make blank knowledge core in arcane crafting terminal: This terminal has some pretty aggressive FPS lag when using with certain animated items, it seems
Arcane inscriber terminal will not allow input of arbitrary recipes from JEI, must be present in-network or in-inventory; Annoying but workable
Encoded filtered essentia tube to blank knowledge core
Add arcane assembler to side of currently in-progress crafting tree
Craft filtered essentia tube: Sub-crafts of essentia filter with existing components. Starts, completes nearly instantly, deducts vis crystals (I think, builder + powered wisp spawner still running was still running)
Tested arcane charger upgrade in arcane crafting terminal, seems to work fine, added to arcane crafter, no displayed vis but seems to work.
Test craft 100 filtered essentia tube, neighboring chunks did drain vis as it worked
Craft aux slurry pump (Making sure dependencies work): No problems

**Comment by realthecaffiend** [comment]
*Also tested partitioning an essentia drive, seems to work in the workbench, and has NBT data on the drive after removing it. I'll run this on our server for a while

**Comment by realthecaffiend** [comment]
Wanted to keep this all in the same place even though this issue is closed, so anyone looking at the fork can check the history haha: I did find a somewhat major problem that is not remediated from the original version.

I checked partitioning an essentia cell, as I thought that was the problem that people ran into (It was something I read on the way by in discord, so, not sure if that was the exact issue or my current problem was). It has NBT data indicating it is filtered, removing and re-adding it to the workbench shows it is filtered, and in a drive it still accepts absolutely any essentia that you happen to throw at it. My essentia build now has to undergo some significant changes, didn't want anyone else with/considering the fork to make the same mistake
